docs/book/v3/application-integration/usage-in-a-laminas-mvc-application.md

```diff
@@ -0,0 +1,181 @@
+# Usage in a laminas-mvc Application
+
+The following example shows _one_ potential use case of laminas-cache within a laminas-mvc based application.
+The example uses a module, a controller and shows the resolving of dependencies of the controller by configuration.
+
+Before starting, make sure laminas-cache is [installed and configured](../installation.md).
+
+> MISSING: **Installation Requirements**
+> laminas-cache is shipped without a specific cache adapter to allow free choice of storage backends and their dependencies.
+> So make sure that the required adapter is installed.
+>
+> The following example used the [filesystem adapter of laminas-cache](../storage/adapter.md#filesystem-adapter):
+>
+> ```bash
+> $ composer require laminas/laminas-cache-storage-adapter-filesystem
+> ```
+
+## Configure Cache
+
+To configure the cache in a laminas-mvc based application, use either application or module configuration (such as `config/autoload/global.php` or `module/Application/config/module.config.php`, respectively), and define the configuration key `caches`.
+
+This example uses the global configuration, e.g. `config/autoload/global.php`:
+
+```php
+return [
+    'caches' => [
+        'default-cache' => [
+            'adapter' => Laminas\Cache\Storage\Adapter\Filesystem::class,
+            'options' => [
+                'cache_dir' => __DIR__ . '/../../data/cache',
+            ],
+        ],
+        // …
+    ],
+    // ...
+];
+```
+
+The factory `Laminas\Cache\Service\StorageCacheAbstractServiceFactory` uses the configuration, searches for the configuration key `caches` and creates the storage adapters using the discovered configuration.
+
+## Create Controller
+
+[Create a controller class](https://docs.laminas.dev/laminas-mvc/quick-start/#create-a-controller) and inject the cache with the interface for all cache storage adapters via the constructor, e.g. `module/Application/Controller/IndexController.php`:
+
+```php
+namespace Application\Controller;
+
+use Laminas\Cache\Storage\StorageInterface;
+use Laminas\Mvc\Controller\AbstractActionController;
+
+final class IndexController extends AbstractActionController
+{
+    public function __construct(
+        private readonly StorageInterface $cache
+    ) {}
+    
+    public function indexAction(): array
+    {
+        if (! $this->cache->hasItem('example')) {
+            $this->cache->addItem('example', 'value');
+        }
+
+        echo $this->cache->getItem('example') // value;
+        
+        // …
+        
+        return [];
+    }
+}
+```
+
+## Register Controller
```

Comment:
do we need this in the cache component? this might change in upcoming mvc versions but wont change here at the same time. 

Id say lets omit this but up to you as you have more experience regarding documentation

Reply:
If we omit this step, we can throw away the entire description.

> …this might change in upcoming mvc versions…

This may be true, but right now there is no plan or movement on this topic. Therefore, it will continue to be explained that way for now.

Reply:
Usually, MVC applications do use factories. I'd say it might be enough when we describe on how to configure the cache component so the `StorageInterface` can be retrieved via the `$container` within a factory.

If upstream projects are using `ConfigAbstractFactory`, they know how to configure that since as far as I know that is not the default suggested way of configuring controllers.
I would simply create an example controller class + example factory rather than referring to the ConfigAbstractFactory.

I can only talk for myself but I am working in a MVC application with ~100 controllers and none of them are configured with the `ConfigAbstractFactory`.

Reply:
At the moment there are no real tools like CLI commands for applications based on laminas-mvc, therefore, everything must be generated by hand. This does not make it easy for beginners.
However, we absolutely need this friendly and simple entrance. And these descriptions under "Application Integrations" must be simple and should use the existing features of the framework / components.

But of course you are right, this again requires a guide that explains the optimization of an application.

Reply:
Some CLI commands can help here to optimise an application by searching and replacing the reflection- and configuration-based factories. This would also be easy to use.